TimeLapseWebHost/FileStore.cs

```diff
@@ -14,54 +20,63 @@ namespace TimeLapseWebHost
     public class FileStore : IFileStore
     {
         private readonly IWebHostEnvironment _env;
-        private readonly string imagesRoot = "Images";
+        private readonly IBlobStorage _blobStorage;
         private const string GifFileName = "MyTimelapse.gif";
 
-        public FileStore(IWebHostEnvironment env)
+        public FileStore(IWebHostEnvironment env, IBlobStorage blobStorage)
         {
             _env = env ?? throw new ArgumentNullException(nameof(env));
+            _blobStorage = blobStorage;
         }
 
         public async Task Create(IFormFile uploadedFile, string id)
         {
-            string userpath = GetUserFolder(id);
+
+            var userBlobContainer = _blobStorage.GetContainer(id);
+            await userBlobContainer.CreateIfNotExistsAsync(Microsoft.WindowsAzure.Storage.Blob.BlobContainerPublicAccessType.Off,null,null);
             var filename = DateTime.Now.Ticks + ".png";
-            var filePath = Path.Combine(userpath, filename);
-            Directory.CreateDirectory(userpath);
-            using (var stream = File.Create(filePath))
+            using (var stream = uploadedFile.OpenReadStream())
             {
-                await uploadedFile.CopyToAsync(stream);
+                await userBlobContainer.CopyToCloud(stream, filename);
             }
         }
 
-        public List<string> GetAll(string id)
+        public async Task<bool> UserHasGif(ClaimsPrincipal user)
         {
-            string userpath = GetUserFolder(id);
-            return Directory.GetFiles(userpath).ToList();
-        }
-
-        public string GetUserFolder(string id)
-        {
-            return Path.Combine(_env.WebRootPath, imagesRoot, id);
-        }
+            var id = user.FindFirst(ClaimTypes.NameIdentifier)?.Value;
+            var userBlobContainer = _blobStorage.GetContainer(id).GetBlobReference(GifFileName);
+            try
```

Comment:
Is de enige manier met try catch? Kan je niet de items in de container listen en checken of ie er tussen zit?